eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -37,82 +37,23 @@ jobs:
         ServiceDirectory: ${{ parameters.ServiceDirectory }}
         BuildTargetingString: ${{ parameters.BuildTargetingString }}
 
-  - job: 'Test'
-    condition: and(succeededOrFailed(), ne(variables['Skip.Test'], 'true'))
-    timeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
-    variables:
-    - template: ../variables/globals.yml
-
-    dependsOn:
-       - 'Build'
-
-    strategy:
-      matrix:
-        Linux_Python27:
-          OSName: 'Linux'
-          OSVmImage: 'ubuntu-16.04'
-          PythonVersion: '2.7'
-          CoverageArg: ''
-        Linux_Python35:
-          OSName: 'Linux'
-          OSVmImage: 'ubuntu-16.04'
-          PythonVersion: '3.5'
-          CoverageArg: ''
-        Linux_Python36:
-          OSName: 'Linux'
-          OSVmImage: 'ubuntu-16.04'
-          PythonVersion: '3.6'
-          CoverageArg: ''
-        Linux_Python37:
-          OSName: 'Linux'
-          OSVmImage: 'ubuntu-16.04'
-          PythonVersion: '3.7'
-          CoverageArg: ''
-        Linux_Python38:
-          OSName: 'Linux'
-          OSVmImage: 'ubuntu-16.04'
-          PythonVersion: '3.8'
-          CoverageArg: ''
-        Windows_Python35:
-          OSName: 'Windows'
-          OSVmImage: 'windows-2019'
-          PythonVersion: '3.5'
-          CoverageArg: ''
-        MacOS_Python27:
-          OSName: 'MacOS'
-          OSVmImage: 'macOS-10.13'
-          PythonVersion: '2.7'
-          CoverageArg: ''
-        Linux_pypy3:
-          OSName: 'Linux'
-          OSVmImage: 'ubuntu-16.04'
-          PythonVersion: 'pypy3'
-          CoverageArg: '--disablecov'
-
-    pool:
-      vmImage: '$(OSVmImage)'
+  - ${{if eq(variables['Build.Reason'], 'PullRequest')}}:
```

Comment:
Is there a way to essentially #ifdef out just part of the matrix instead of splitting up the entire test section? 

Not sure if this actually works but I was thinking about something like:
```
  strategy:
      matrix:
        Linux_Python27:
          OSName: 'Linux'
          OSVmImage: 'ubuntu-16.04'
          PythonVersion: '2.7'
          CoverageArg: ''
        Linux_Python38:
          OSName: 'Linux'
          OSVmImage: 'ubuntu-16.04'
          PythonVersion: '3.8'
          CoverageArg: ''
    - ${{if ne(variables['Build.Reason'], 'PullRequest')}}:
        Linux_Python35:
          OSName: 'Linux'
          OSVmImage: 'ubuntu-16.04'
          PythonVersion: '3.5'
          CoverageArg: ''
        Linux_Python36:
          OSName: 'Linux'
          OSVmImage: 'ubuntu-16.04'
          PythonVersion: '3.6'
          CoverageArg: ''
        Linux_Python37:
          OSName: 'Linux'
          OSVmImage: 'ubuntu-16.04'
          PythonVersion: '3.7'
          CoverageArg: ''
        Windows_Python35:
          OSName: 'Windows'
          OSVmImage: 'windows-2019'
          PythonVersion: '3.5'
          CoverageArg: ''
        MacOS_Python27:
          OSName: 'MacOS'
          OSVmImage: 'macOS-10.13'
          PythonVersion: '2.7'
          CoverageArg: ''
        Linux_pypy3:
          OSName: 'Linux'
          OSVmImage: 'ubuntu-16.04'
          PythonVersion: 'pypy3'
          CoverageArg: '--disablecov'
```

Reply:
Updated as per this suggestion

Reply:
Did you do that in a new PR?

Reply:
Yup. I have opened PR https://github.com/Azure/azure-sdk-for-python/pull/9031 for this change.